### ISSUE TITLE
Istio-token permissions problem on EKS - failed to fetch token from file

### ISSUE THREAD

**Issue by williamaronli** [istio/istio#31740]
**Bug description**
From one of our customer issue: https://discuss.istio.io/t/istio-token-permissions-problem-on-eks-failed-to-fetch-token-from-file-open-var-run-secrets-tokens-istio-token-permission-denied/10028

I have created istio enabled service on EKS 1.19 with istio 1.9 The service was created but the proxy was not able to authenticate. I have entered the proxy pod and have found that ./var/run/secrets/tokens/istio-token is not accessible by the istio-proxy account. The token has root permissions:lrwxrwxrwx 1 root root 18 Mar 16 20:06 ./var/run/secrets/tokens/istio-token → …data/istio-token. The proxy has “failed to fetch token from file: open ./var/run/secrets/tokens/istio-token: permission denied” message in the logs. Is there a way to fix the problem?

I have recreated cluster with eksctl and istio works now. I suspect that upgrade to k8s 1.19 created the problem.

[ ] Docs
[ ] Installation
[X] Networking
[ ] Performance and Scalability
[ ] Extensions and Telemetry
[X] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure
[ ] Upgrade

**Expected behavior**

**Steps to reproduce the bug**

**Version** (include the output of `istioctl version --remote` and `kubectl version --short` and `helm version --short` if you used Helm)

**How was Istio installed?**

**Environment where the bug was observed (cloud vendor, OS, etc)**

Additionally, please consider running `istioctl bug-report` and attach the generated cluster-state tarball to this issue.
Refer [cluster state archive](http://istio.io/help/bugs/#generating-a-cluster-state-archive) for more details.

**Comment by howardjohn** [comment]
Likely broken by https://github.com/istio/istio/blob/6107d132a8bb8793ccd7f6a42e2e1ca2778c4ebf/manifests/charts/istiod-remote/files/injection-template.yaml#L493. Maybe EKS broke/did not apply the fix that we rely on?

**Comment by howardjohn** [comment]
the upstream fix is not conditionally enabled so I don't see how/why it would not work for EKS: https://github.com/kubernetes/kubernetes/commit/3db7275b549559696c42c0b5f51c9a2397e9571d

**Comment by kirankumar15** [comment]
Even I am seeing the same issue for EKS 1.19 with Istio 1.9.2. It worked when I installed istio with the following option `--set values.pilot.env.ENABLE_LEGACY_FSGROUP_INJECTION=true`

`istioctl install --set profile=demo -y --set values.pilot.env.ENABLE_LEGACY_FSGROUP_INJECTION=true`

**Comment by esnible** [comment]
I can confirm that after I upgraded to Kubernetes v1.19.9+IKS (IBM Cloud), I started getting the token errors.  The errors did not occur until the pods restarted.
I can confirm that using `--set values.pilot.env.ENABLE_LEGACY_FSGROUP_INJECTION=true` makes the problem go away.

My K8s workers were 1.18 and 1.17.  Based on the comments on Istio Discuss, I upgraded my workers to 1.19 and seem to no longer needed the legacy injection.
